exporters/otlp/otlp_exporter_benchmark.cc

```diff
@@ -0,0 +1,164 @@
+                      grpc::CompletionQueue *) override
+  {
+    return nullptr;
+  }
+
+  grpc::ClientAsyncResponseReaderInterface<proto::collector::trace::v1::ExportTraceServiceResponse>
+      *PrepareAsyncExportRaw(grpc::ClientContext *,
+                             const proto::collector::trace::v1::ExportTraceServiceRequest &,
+                             grpc::CompletionQueue *) override
+  {
+    return nullptr;
+  }
+};
+
+// OtlpExporterTestPeer is a friend class of OtlpExporter
+class OtlpExporterTestPeer
+{
+public:
+  std::unique_ptr<sdk::trace::SpanExporter> GetExporter()
+  {
+    auto mock_stub = new FakeServiceStub();
+    std::unique_ptr<proto::collector::trace::v1::TraceService::StubInterface> stub_interface(
+        mock_stub);
+    return std::unique_ptr<sdk::trace::SpanExporter>(
+        new exporter::otlp::OtlpExporter(std::move(stub_interface)));
+  }
+};
+
+// Helper function to create empty spans
+void CreateEmptySpans(std::array<std::unique_ptr<sdk::trace::Recordable>, kBatchSize> &recordables)
+{
+  for (int i = 0; i < kBatchSize; i++)
+  {
+    auto recordable = std::unique_ptr<sdk::trace::Recordable>(new Recordable);
+    recordables[i]  = std::move(recordable);
+  }
+}
+
+// Helper function to create sparse spans
+void CreateSparseSpans(std::array<std::unique_ptr<sdk::trace::Recordable>, kBatchSize> &recordables)
+{
+  for (int i = 0; i < kBatchSize; i++)
+  {
+    auto recordable = std::unique_ptr<sdk::trace::Recordable>(new Recordable);
+
+    recordable->SetIds(kTraceId, kSpanId, kParentSpanId);
+    recordable->SetName("TestSpan");
+    recordable->SetStartTime(core::SystemTimestamp(std::chrono::system_clock::now()));
+    recordable->SetDuration(std::chrono::nanoseconds(10));
+
+    recordables[i] = std::move(recordable);
+  }
+}
+
+// Helper function to create dense spans
+void CreateDenseSpans(std::array<std::unique_ptr<sdk::trace::Recordable>, kBatchSize> &recordables)
+{
+  for (int i = 0; i < kBatchSize; i++)
+  {
+    auto recordable = std::unique_ptr<sdk::trace::Recordable>(new Recordable);
+
+    recordable->SetIds(kTraceId, kSpanId, kParentSpanId);
+    recordable->SetName("TestSpan");
+    recordable->SetStartTime(core::SystemTimestamp(std::chrono::system_clock::now()));
+    recordable->SetDuration(std::chrono::nanoseconds(10));
+
+    for (int i = 0; i < kNumAttributes; i++)
+    {
+      recordable->SetAttribute("int_key_" + i, static_cast<int64_t>(i));
+      recordable->SetAttribute("str_key_" + i, "string_val_" + i);
+      recordable->SetAttribute("bool_key_" + i, true);
+    }
+
+    recordables[i] = std::move(recordable);
+  }
+}
+
+// ------------------------------ Benchmark tests ------------------------------
+
+// Benchmark Export() with empty spans
+void BM_OtlpExporterEmptySpans(benchmark::State &state)
+{
+  std::unique_ptr<OtlpExporterTestPeer> testpeer(new OtlpExporterTestPeer());
+  auto exporter = testpeer->GetExporter();
+
+  for (auto _ : state)
```

Comment:
Might want to look into using [State::KeepRunningBatch()](https://github.com/google/benchmark/blob/master/include/benchmark/benchmark.h#L494) instead of [State::KeepRunning()](https://github.com/google/benchmark/blob/master/include/benchmark/benchmark.h#L482), because of reasons stated in [this PR](https://github.com/google/benchmark/pull/521#issue-166342372).

Reply:
@pyohannes any thoughts on this?

Reply:
I'm not entirely clear on the benefits of this, other than speeding up the tests. Is that something that's important to us?